libraries/c_sdk/standard/mqtt/src/iot_mqtt_serialize.c

```diff
@@ -302,1638 +146,37 @@ static uint16_t _nextPacketIdentifier( void )
-         * pPacketSize output parameter. */
-        subscriptionPacketSize += 1 + _remainingLengthEncodedSize( subscriptionPacketSize );
-        *pPacketSize = subscriptionPacketSize;
-    }
-
-    return status;
-}
-
-/*-----------------------------------------------------------*/
-
-uint8_t _IotMqtt_GetPacketType( void * pNetworkConnection,
-                                const IotNetworkInterface_t * pNetworkInterface )
-{
-    uint8_t packetType = 0xff;
-
-    /* The MQTT packet type is in the first byte of the packet. */
-    ( void ) _IotMqtt_GetNextByte( pNetworkConnection,
-                                   pNetworkInterface,
-                                   &packetType );
-
-    return packetType;
-}
-
-/*-----------------------------------------------------------*/
-
-size_t _IotMqtt_GetRemainingLength( void * pNetworkConnection,
-                                    const IotNetworkInterface_t * pNetworkInterface )
-{
-    uint8_t encodedByte = 0;
-    size_t remainingLength = 0, multiplier = 1, bytesDecoded = 0, expectedSize = 0;
-
-    /* This algorithm is copied from the MQTT v3.1.1 spec. */
-    do
-    {
-        if( multiplier > 2097152 ) /* 128 ^ 3 */
-        {
-            remainingLength = MQTT_REMAINING_LENGTH_INVALID;
-            break;
-        }
-        else
-        {
-            if( _IotMqtt_GetNextByte( pNetworkConnection,
-                                      pNetworkInterface,
-                                      &encodedByte ) == true )
-            {
-                remainingLength += ( encodedByte & 0x7F ) * multiplier;
-                multiplier *= 128;
-                bytesDecoded++;
-            }
-            else
-            {
-                remainingLength = MQTT_REMAINING_LENGTH_INVALID;
-                break;
-            }
-        }
-    } while( ( encodedByte & 0x80 ) != 0 );
-
-    /* Check that the decoded remaining length conforms to the MQTT specification. */
-    if( remainingLength != MQTT_REMAINING_LENGTH_INVALID )
-    {
-        expectedSize = _remainingLengthEncodedSize( remainingLength );
-
-        if( bytesDecoded != expectedSize )
-        {
-            remainingLength = MQTT_REMAINING_LENGTH_INVALID;
-        }
-        else
-        {
-            /* Valid remaining length should be at most 4 bytes. */
-            IotMqtt_Assert( bytesDecoded <= 4 );
-        }
-    }
-    else
-    {
-        EMPTY_ELSE_MARKER;
-    }
-
-    return remainingLength;
-}
-
-/*-----------------------------------------------------------*/
-
-IotMqttError_t _IotMqtt_SerializeConnect( const IotMqttConnectInfo_t * pConnectInfo,
-                                          uint8_t ** pConnectPacket,
-                                          size_t * pPacketSize )
+void _IotMqtt_PublishSetDup( uint8_t * pPublishPacket,
+                             uint8_t * pPacketIdentifierHigh,
+                             uint16_t * pNewPacketIdentifier )
```

Comment:
As I recall, the plan is to get rid of this file later as part of a clean up task right?

Reply:
Can we rename the file to indicate this is not serializer any more?

Reply:
I would get rid of this file if possible to avoid confusion.